exec/cnex/lib/os_posix.c

```diff
@@ -0,0 +1,197 @@
+// The following is needed for kill() on Posix platforms.
+#define _POSIX_C_SOURCE 200809L
+
+#include <assert.h>
+#include <errno.h>
+#include <signal.h>
+#include <stdlib.h>
+#include <sys/param.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+
+#include "cell.h"
+#include "enums.h"
+#include "exec.h"
+#include "number.h"
+#include "object.h"
+#include "stack.h"
+
+
+
+static Array *g_children;
+
+typedef struct tagTProcessHandle {
+    pid_t proc;
+} TProcessHandle;
+
+
+
+static TProcessHandle *check_process(TExecutor *exec, Object *pp)
+{
+    if (pp == NULL || pp->ptr == NULL || ((TProcessHandle*)pp->ptr)->proc == 0) {
+        exec->rtl_raise(exec, "OsException.InvalidProcess", "", BID_ZERO);
+        return NULL;
+    }
+    return pp->ptr;
+}
+
+void object_releaseProcessObject(Object *o)
+{
+    if (o != NULL) {
+        assert(o->refcount > 0);
+        o->refcount--;
+
+        if (o->refcount <= 0) {
+            if (o->ptr != NULL) {
+                if (((TProcessHandle*)o->ptr)->proc != 0) {
+                    close(((TProcessHandle*)o->ptr)->proc);
```

Comment:
This is not how you clean up pids. A pid is not a file handle, and you'll end up closing some random file handle here.

Reply:
What should I do here?  When a process is being cleaned up, should I call kill(), or is no operation whatsoever needed here?